fdbserver/Status.actor.cpp

```diff
@@ -3053,6 +3053,22 @@ ACTOR Future<StatusReply> clusterGetStatus(
 	state WorkerDetails csWorker; // ConsistencyScan worker
 
 	try {
+
+		state JsonBuilderObject statusObj;
+
+		// Get EKP Health
+		if (db->get().client.encryptKeyProxy.present()) {
+			EKPHealthStatus status = wait(GetEncryptCipherKeys<ServerDBInfo>::getEKPHealthStatus(db));
+			JsonBuilderObject _statusObj;
+			_statusObj["kms_is_healthy"] = status.canConnectToKms;
+			_statusObj["ekp_is_healthy"] = status.canConnectToEKP;
+			statusObj["encryption_at_rest"] = _statusObj;
+			// TODO: In this scenario we should see if we can fetch any status fields that don't depend on encryption
+			if (!status.canConnectToKms || !status.canConnectToEKP) {
+				return StatusReply(statusObj.getJson());
```

Comment:
If the KMS is just flaky and doesn't completely take down FDB, maybe we should show the rest of the status json?

Reply:
see above

Reply:
There is a follow up task to fetch fields that are not dependent on encryption when kms is down, that should cover it